src/runtime/profiling.cc

```diff
@@ -822,25 +823,37 @@ PackedFunc WrapTimeEvaluator(PackedFunc pf, Device dev, int number, int repeat,
         f_preproc.CallPacked(args, &temp);
       }
       double duration_ms = 0.0;
-
+      std::vector<double> results;
+      int64_t final_number = number;
       do {
+        results.clear();
         if (duration_ms > 0.0) {
-          number = static_cast<int>(std::max((min_repeat_ms / (duration_ms / number) + 1),
-                                             number * 1.618));  // 1.618 is chosen by random
+          final_number =
+              static_cast<int>(std::max((min_repeat_ms / (duration_ms / final_number) + 1),
+                                        final_number * 1.618));  // 1.618 is chosen by random
         }
+        results.resize(final_number);
 
-        Timer t = Timer::Start(dev);
-        // start timing
-        for (int i = 0; i < number; ++i) {
+        for (int i = 0; i < final_number; ++i) {
+          // start timing
+          Timer t = Timer::Start(dev);
```

Comment:
The reason we have the timer outside the loop is so that we can accurately time very small duration functions. We can amortize the inaccuracy in the timer by running the small duration programs many times under a single timer and then dividing the total time by the number of runs.

Reply:
Reverted